cli/src/wasm.rs

```diff
@@ -116,3 +123,23 @@ pub fn compile_language_to_wasm(language_dir: &Path, force_docker: bool) -> Resu
 
     Ok(())
 }
+
+fn get_emcc_path() -> Result<PathBuf> {
+    let emcc_bin;
+    if cfg!(windows) {
+        emcc_bin = "emcc.bat";
+    } else {
+        emcc_bin = "emcc";
+    };
+    let emcc_which = which(emcc_bin);
```

Comment:
Could you explain the reason for using the `which` crate here, compared just checking the result of `Command::new`, since it already looks up the command on your `PATH`?

Reply:
Built-in Rust's `Command` apparently has a buggy behavior.


Running tree-sitter build-wasm with the following code
```rs
fn get_emcc_path() -> Result<String> {
    let emcc_bin = if cfg!(windows) {
                        "emcc.bat"
                    } else {
                        "emcc"
                    };
    if Command::new(emcc_bin).output().is_ok() {
        return Ok(emcc_bin.to_string());
    }
    return Error::err("emcc binary not found".to_string());
}
```

Results in this error:
```
emcc command failed - C:\Users\aminy\AppData\Local\emsdk\python\3.9.2-1_64bit\python.exe: can't open file 'C:\Users\aminy\Documents\GitHub\Dream\tree-sitter-python\emcc.py': [Errno 2] No such file or directory
```

Reply:
Interesting. Thanks for explaining.

Reply:
You're welcome